Code/Framework/AzFramework/AzFramework/TargetManagement/TargetManagementComponent.cpp

```diff
@@ -682,105 +699,83 @@ namespace AzFramework
     {
         while (!m_stopRequested)
         {
-            if (m_networkImpl->m_gridMate)
+            AZ_PROFILE_SCOPE(AzFramework, "TargetManager::Tick");
+
+            // Send
+            size_t maxMsgsToSend = m_outbox.size();
+            const TargetInfo ti = GetMyTargetInfo();
+            for (size_t iSend = 0; iSend < maxMsgsToSend; ++iSend)
             {
-                AZ_PROFILE_SCOPE(AzFramework, "TargetManager::Tick");
-                if (!m_networkImpl->m_session && !m_networkImpl->m_gridSearch)
-                {
-                    if (AZStd::chrono::system_clock::now() > m_reconnectionTime)
-                    {
-                        JoinNeighborhood();
-                    }
-                }
+                // Send outbound data
+
+                // Lock outbox to prevent a read/write race
+                m_outboxMutex.lock();
 
+                uint8_t* outBuffer = (uint8_t*)m_outbox.front().second.data();
+                const size_t totalSize = m_outbox.front().second.size();
+                size_t outSize = m_outbox.front().second.size();
+                while (outSize > 0)
                 {
-                    AZ_PROFILE_SCOPE(AzFramework, "TargetManager::Tick Gridmate");
-                    m_networkImpl->m_gridMate->Update();
-                    if (m_networkImpl->m_session && m_networkImpl->m_session->GetReplicaMgr())
-                    {
-                        m_networkImpl->m_session->GetReplicaMgr()->Unmarshal();
-                        m_networkImpl->m_session->GetReplicaMgr()->UpdateFromReplicas();
-                        m_networkImpl->m_session->GetReplicaMgr()->UpdateReplicas();
-                        m_networkImpl->m_session->GetReplicaMgr()->Marshal();
-                    }
+                    size_t bufferSize = AZStd::min(outSize, aznumeric_cast<size_t>(Neighborhood::NeighborBufferSize));
+                    AzFrameworkPackets::TargetMessage tmPacket;
+                    tmPacket.SetPersistentId(ti.GetPersistentId());
+                    tmPacket.SetSize(aznumeric_cast<uint32_t>(totalSize));
+                    Neighborhood::NeighborMessageBuffer encodingBuffer;
+                    encodingBuffer.CopyValues(outBuffer + (totalSize - outSize), bufferSize);
+                    tmPacket.SetMessageBuffer(encodingBuffer);
+                    outSize -= bufferSize;
+
+                    m_networkInterface->SendReliablePacket(
+                        static_cast<AzNetworking::ConnectionId>(m_settings->m_lastTarget.GetNetworkId()), tmPacket);          
                 }
 
-                if (m_networkImpl->m_session)
+                m_outbox.pop_front();
+                m_outboxMutex.unlock();
+            }
+
+            AZStd::this_thread::sleep_for(AZStd::chrono::milliseconds(50));
```

Comment:
magic number and sleep. both are red flags ;)

move magic number to named and commented const, consider doing something other than sleeping here.

Reply:
The sleep isn't particularly bad given it's in a side thread. That being said, it really shouldn't be necessary considering we're gating with a size check.